File: DoohSamikshya.Model/Application/Inv/Screen.cs
using DoohSamikshya.Model.Shared.Enum;
using System.Text.Json.Serialization;

namespace DoohSamikshya.Model.Application.Inv
{
    public class Screen
    {
        public int Id { get; set; }
        public int TenantId { get; set; }
        public string Name { get; set; } = null!;
        public string Location { get; set; } = null!;
        public string Resolution { get; set; } = "1920x1080";
        public string? ResolutionName { get; set; }

        [JsonIgnore]
        public string? Tag { get; set; }

        [JsonPropertyName("tag")]
        public List<string>? TagList
        {
            get
            {
                if (string.IsNullOrWhiteSpace(Tag)) return new List<string>();
                return Tag.Split(',', StringSplitOptions.RemoveEmptyEntries)
                          .Select(t => t.Trim())
                          .ToList();
            }
            set
            {
                if (value == null || value.Count == 0)
                    Tag = null;
                else
                    Tag = string.Join(",", value.Select(t => t.Trim()));
            }
        }

        public ScreenOrientation Orientation { get; set; } = ScreenOrientation.Landscape;
        public string? OrientationName { get; set; }
        public ScreenStatus Status { get; set; } = ScreenStatus.Active;
        public string? StatusName { get; set; }
        public bool IsDeleted { get; set; } = false;
        public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
        public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
        public int? CreatedBy { get; set; }
        public int UpdatedBy { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }
        public int? DeletedBy { get; set; }
        public List<ScreenOperatingHour> OperatingHour { get; set; } = new();
    }

    public class ScreenFilter
    {
        public string? Search { get; set; }
        public ScreenStatus? Status { get; set; }
        public ScreenOrientation? Orientation { get; set; }
        public int Offset { get; set; } = 0;     
        public int PageSize { get; set; } = 10;
    }

    public class ScreenResponse
    {
        public int TotalRows { get; set; }  
        public List<Screen>? Data { get; set; }
    }
}

File: DoohFrontend/src/app/screens/model/Screen.ts
import { ScreenStatus, ScreenOrientation, ScreenResolution } from "./ScreenEnum";  

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
  errors: any;
}

export class Screens {
  id: number;
  tenantId: number;
  name: string;
  location: string;
  resolution: string;
  tag: string[] | null;

  orientation: ScreenOrientation;
  status: ScreenStatus;

  isDeleted: boolean;

  createdAt?: string;
  updatedAt?: string;
  createdBy: number | null;
  updatedBy: number;
  deletedAt?: string | null;
  deletedBy?: number | null;

  operatingHour: ScreenOperatingHour[];

  constructor() {
    this.id = 0;
    this.tenantId = 0;
    this.name = '';
    this.location = '';
    this.resolution = '';

    this.tag = null;

    this.orientation = ScreenOrientation.Landscape;
    this.status = ScreenStatus.Active;      

    this.isDeleted = false;

    this.createdAt = new Date().toISOString();
    this.updatedAt = new Date().toISOString();
    this.createdBy = null;
    this.updatedBy = 0;
    this.deletedAt = null;
    this.deletedBy = null;

    this.operatingHour = [];
  }
}

export interface DropdownItemScreen {
    id: number;
    name: string;
}

export class ScreenOperatingHour {
  id: number;
  screenId: number;

  startTime: string;
  endTime: string;
  dayOfWeek: number;
  avgAudienceCount: number;

  isDeleted: boolean;

  createdAt?: string;
  updatedAt?: string;
  createdBy: number | null;
  updatedBy: number;
  deletedAt: string | null;
  deletedBy?: number | null;
  //isEveryday?: boolean;

  constructor() {
    this.id = 0;
    this.screenId = 0;

    this.startTime = '00:00:00';
    this.endTime = '00:00:00';
    this.dayOfWeek = 0;
    this.avgAudienceCount = 0;

    this.isDeleted = false;
   // this.isEveryday = false;
    this.createdAt = new Date().toISOString();
    this.updatedAt = new Date().toISOString();
    this.createdBy = null;
    this.updatedBy = 0;
    this.deletedAt = null;
    this.deletedBy = null;
  }
}

export interface OperatingHourSlot {
  id: string;
  selectedDays: number[];
  startTime: string;
  endTime: string;
  avgAudienceCount: number;
}

export interface ScreenFilter {
  search?: string;
  status?: ScreenStatus | null;
  orientation?: ScreenOrientation | null;
  offset?: number;
  pageSize?: number;
}

export interface ScreenResponse {
  totalRows: number;
  data: Screens[];
}

export class ScreenInsert {
  tenantId: number;
  name: string;
  location: string;
  //resolution: string;
  tag: string[] | null;
  resolution : ScreenResolution;
  orientation: ScreenOrientation;
  status: ScreenStatus;

  createdBy: number;

  operatingHour: ScreenOperatingHour[];

  constructor() {
    this.tenantId = 0;
    this.name = '';
    this.location = '';
    this.resolution = ScreenResolution.R1920x1080;

    this.tag = null;

    this.orientation = ScreenOrientation.Landscape;
    this.status = ScreenStatus.Active;

    this.createdBy = 1;

    this.operatingHour = [];
  }
}

export class ScreenUpdate {
  id: number;

  name: string;
  location: string;
  resolution : ScreenResolution;
  tag: string[] | null;

  orientation: ScreenOrientation;
  status: ScreenStatus;

  updatedBy: number;

  operatingHour: ScreenOperatingHour[];

  constructor() {
    this.id = 0;

    this.name = '';
    this.location = '';
    this.resolution = ScreenResolution.R1920x1080;

    this.tag = null;

    this.orientation = ScreenOrientation.Landscape;
    this.status = ScreenStatus.Active;

    this.updatedBy = 1;

    this.operatingHour = [];
  }
}





